### PR TITLE
Minishift support

### PR DESCRIPTION
**Pull request by l0rd**
Added minishift as a new installation platform. 

I have also changed the default namespace name to `che` because we cannot create namespaces prefixed with `kube-` on OpenShift.

### CODE REVIEW

**Comment by azatsarynnyy** [comment]
After running `server:start -p minishift -a operator`, downloading images phase has been stopped by timeout:
```
  ✔ ✈️  Minishift preflight checklist
    ✔ Verify if oc is installed...done.
    ✔ Verify if minishift is installed...done.
    ✔ Verify if minishift is running...done.
    ✔ Retrieving minishift IP and domain for routes URLs...192.168.99.152.nip.io.
  ✔ 🏃‍  Running the Che Operator
    ✔ Create Namespace (che)...done.
    ✔ Create ServiceAccount che-operator...done.
    ✔ Create RoleBinding che-operator in namespace che...done.
    ✔ Create ConfigMap che-operator in namespace che...done.
    ✔ Patch ConfigMap che-operator in namespace che...done.
    ✔ Delete Pod che-operator in namespace che if it exist...the Pod doesn't exist. Skipping.
    ✔ Waiting 5 seconds for the new ServiceAccount and RoleBindings to be flushed...done.
    ✔ Create che-operator Pod in namespace che...done.
  ❯ ✅  Post installation checklist
    ✔ PostgreSQL pod bootstrap
      ✔ scheduling...done.
      ✔ downloading images...done.
      ✔ starting...done.
    ❯ Keycloak pod bootstrap
      ✔ scheduling...done.
      ✖ downloading images
        → ERR_TIMEOUT
        starting
      Che pod bootstrap
      Retrieving Che Server URL
      Che status check
Error: ERR_TIMEOUT
    at KubeHelper.<anonymous> (~/github/l0rd/chectl/src/api/kube.ts:217:11)
    at Generator.next (<anonymous>)
    at fulfilled (~/github/l0rd/chectl/src/api/kube.ts:14:58)
    at <anonymous>
```

After the second run, I've got:
```
  ✔ ✈️  Minishift preflight checklist
    ✔ Verify if oc is installed...done.
    ✔ Verify if minishift is installed...done.
    ✔ Verify if minishift is running...done.
    ✔ Retrieving minishift IP and domain for routes URLs...192.168.99.152.nip.io.
  ✔ 🏃‍  Running the Che Operator
    ✔ Create Namespace (che)...It already exist.
    ✔ Create ServiceAccount che-operator...It already exist.
    ✔ Create RoleBinding che-operator in namespace che...It already exist.
    ✔ Create ConfigMap che-operator in namespace che...It already exist.
    ✔ Patch ConfigMap che-operator in namespace che...done.
    ✔ Delete Pod che-operator in namespace che if it exist...done.
    ✔ Waiting 5 seconds for the new ServiceAccount and RoleBindings to be flushed...done.
    ✔ Create che-operator Pod in namespace che...done.
  ❯ ✅  Post installation checklist
    ❯ PostgreSQL pod bootstrap
      ✖ scheduling
        → ERR_UNEXPECTED_PHASE: Running (Pending expected)
        downloading images
        starting
      Keycloak pod bootstrap
      Che pod bootstrap
      Retrieving Che Server URL
      Che status check
Error: ERR_UNEXPECTED_PHASE: Running (Pending expected) 
    at KubeHelper.<anonymous> (~/github/l0rd/chectl/src/api/kube.ts:229:17)
    at Generator.next (<anonymous>)
    at fulfilled (~/github/l0rd/chectl/src/api/kube.ts:14:58)
    at <anonymous>
```

**Comment by benoitf** [comment]
there is an existing issue I think for the running/pending
(you need as workaround to do server:stop and then server:start)

**Comment by benoitf** [comment]
https://github.com/che-incubator/chectl/issues/48

**Comment by azatsarynnyy** [comment]
`server:stop`
```
  ✔ Verify if namespace che exist
  ✔ Verify if helm is installed
  ✖ Stopping Che server
    → Error: pods is forbidden: User "developer" cannot list pods in the namespace "kube-system": no RBAC policy matched
Error: Command failed: /bin/sh -c helm delete che --purge
Error: pods is forbidden: User "developer" cannot list pods in the namespace "kube-system": no RBAC policy matched
    at makeError (~/github/l0rd/chectl/node_modules/execa/index.js:174:9)
    at Promise.all.then.arr (~/github/l0rd/chectl/node_modules/execa/index.js:278:16)
    at <anonymous>
```

**Comment by azatsarynnyy** [comment]
Che Server has been started successfully after increasing the timeout [here](https://github.com/che-incubator/chectl/blob/master/src/api/kube.ts#L208) to 300000 ms.
```
  ✔ ✈️  Minishift preflight checklist
    ✔ Verify if oc is installed...done.
    ✔ Verify if minishift is installed...done.
    ✔ Verify if minishift is running...done.
    ✔ Retrieving minishift IP and domain for routes URLs...192.168.99.153.nip.io.
  ✔ 🏃‍  Running the Che Operator
    ✔ Create Namespace (che)...done.
    ✔ Create ServiceAccount che-operator...done.
    ✔ Create RoleBinding che-operator in namespace che...done.
    ✔ Create ConfigMap che-operator in namespace che...done.
    ✔ Patch ConfigMap che-operator in namespace che...done.
    ✔ Delete Pod che-operator in namespace che if it exist...the Pod doesn't exist. Skipping.
    ✔ Waiting 5 seconds for the new ServiceAccount and RoleBindings to be flushed...done.
    ✔ Create che-operator Pod in namespace che...done.
  ✔ ✅  Post installation checklist
    ✔ PostgreSQL pod bootstrap
      ✔ scheduling...done.
      ✔ downloading images...done.
      ✔ starting...done.
    ✔ Keycloak pod bootstrap
      ✔ scheduling...done.
      ✔ downloading images...done.
      ✔ starting...done.
    ✔ Che pod bootstrap
      ✔ scheduling...done.
      ✔ downloading images...done.
      ✔ starting...done.
    ✔ Retrieving Che Server URL...http//"che-che.192.168.99.153.nip.io
    ✔ Che status check
```
But URL is incorrect a bit :slightly_smiling_face:

File: src/api/openshift.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+ * SPDX-License-Identifier: EPL-2.0
+ **********************************************************************/
+// tslint:disable:object-curly-spacing
+
+import execa = require('execa')
+
+export class OpenShiftHelper {
+  async getHostByRouteName(routeName: string, namespace = ''): Promise<string> {
+    const command = 'oc'
+    const args = ['get', 'route', '--namespace', namespace, '-o', `jsonpath="{range .items[?(.metadata.name=='${routeName}')]}{.spec.host}{end}`]
```
Comment (azatsarynnyy) [inline src/api/openshift.ts]: `jsonpath="` - extra `"`
`range .items` - extra space

File: src/commands/server/start.ts [hunk 2]
```diff
@@ -167,7 +175,15 @@ export default class Start extends Command {
...
       title: 'Retrieving Che Server URL',
       task: async (ctx: any, task: any) => {
-        ctx.cheURL = await che.cheURLByIngress(ingressName, flags.chenamespace)
+        if (flags.platform === 'minikube') {
+          ctx.cheURL = await che.cheURLByIngress(ingressName, flags.chenamespace)
+        } else if (flags.platform === 'minishift') {
+          const os = new OpenShiftHelper()
+          const hostname = await os.getHostByRouteName(ingressName, flags.chenamespace)
+          const protocol = flags.tls ? 'https' : 'http'
+          ctx.cheURL = `${protocol}//${hostname}`
```
Comment (azatsarynnyy) [inline src/commands/server/start.ts]: `${protocol:}//${hostname}` - missing `:`

**Comment by l0rd** [comment]
Thank you @azatsarynnyy for your review. I have fixed the problems with the URL and increased the timeout to 5min when waiting for the image to be downloaded (but for a better UX we should fix #47 and #48 in a proper way).
 
For the stop command I won't fix it here since it's not related to the platform (minishift) but more to the installer (operator). I have opened #56 anyway.

**Comment by l0rd** [comment]
@benoitf are you ok if I merge it or do you want to review it?

**Comment by benoitf** [comment]
you can merge

**Pull request merged**